### PR TITLE
Dependency guard jsonpb and protojson packages

### PR DESCRIPTION
**Pull request by ymesika**
**Please provide a description of this PR:**

Added a dependency guard to CI to make sure we have the use of jsonpb (deprecated but we still need to use this one) and protojson in a single wrapper.
Asking developers to use the protomarshal instead.

As suggested in: https://github.com/istio/istio/pull/42722#issue-1525901856

### CODE REVIEW

File: pkg/config/xds/xds.go
Comment (howardjohn) [inline pkg/config/xds/xds.go:82]: does missing origname matter?

Comment (howardjohn) [reply pkg/config/xds/xds.go:82]: Tests pass so maybe not....

**Review by howardjohn — APPROVED** [review]
nice! 

Note: the common/ files are from istio/common-files, we need to copy it over there once its merged. Its good to merge here first though, like you did, so we get CI checking it

**Review by lei-tang — APPROVED** [review]

File: pkg/config/xds/xds.go
Comment (ymesika) [reply pkg/config/xds/xds.go:82]: `MarshalProtoNames` has OrigName true.
It's your impl :)

**Review by hzxuzhonghu — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by linsun** [comment]
@therealmitchconnors @zirain @irisdingbj can you pls review/approve?

**Comment by zirain** [comment]
> @therealmitchconnors @zirain @irisdingbj can you pls review/approve?

I'm not one of UX maintainer, cc @hanxiaop

**Review by therealmitchconnors — APPROVED** [review]

**Comment by ericvn** [comment]
/cherry-pick experimental-ambient

Not sure if this will cp successfully.

**Pull request merged**
